### PR TITLE
feat: Parse command line arguments in settlement report job

### PR DESCRIPTION
**Pull request by JesperGomes**
# Description

This PR takes command line arguments, validates them, and creates a SettlementReportArgs object. 

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [x] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/calculation_engine/package/settlement_report_job/args_helper.py
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/settlement_report_job/args_helper.py:1]: How about creating an infrastructure subpackage including this and the other modules concerning infra?

File: source/databricks/calculation_engine/package/settlement_report_job/logger.py
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/settlement_report_job/logger.py:1]: Should we create a story about moving this to a shared package?

File: source/databricks/calculation_engine/package/settlement_report_job/settlement_report.py [hunk 1]
```diff
@@ -12,17 +12,83 @@
...
+    applicationinsights_connection_string = os.getenv(
+        "APPLICATIONINSIGHTS_CONNECTION_STRING"
+    )
+
+    start_with_deps(
+        applicationinsights_connection_string=applicationinsights_connection_string
+    )
+
+
+def start_with_deps(
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/settlement_report_job/settlement_report.py:45]: I know I'm the originator of this terrible mess. Let's add to the story about a logging package that we should extract the logging concerns of this function and provide more generically in the logging package.

File: source/databricks/calculation_engine/package/settlement_report_job/time_series_points.py
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/settlement_report_job/time_series_points.py:1]: time_series_points_factory.py? time_series_points_repository.py?

File: source/databricks/calculation_engine/tests/settlement_report_job/conftest.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+from package.settlement_report_job.calculation_type import CalculationType
+from package.settlement_report_job.settlement_report_args import SettlementReportArgs
+
+
+@pytest.fixture(scope="session")
+def any_settlement_report_args() -> SettlementReportArgs:
+    return SettlementReportArgs(
+        report_id=str(uuid.uuid4()),
+        period_start=datetime(2018, 1, 31, 23, 0, 0),
+        period_end=datetime(2018, 1, 31, 23, 0, 0),
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/settlement_report_job/conftest.py:28]: 0 day period?

File: source/databricks/calculation_engine/tests/settlement_report_job/test_settlement_report_args.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+from package.settlement_report_job.settlement_report import (
+    parse_job_arguments,
+    parse_command_line_arguments,
+)
+
+from package.settlement_report_job.environment_variables import EnvironmentVariable
+from package.settlement_report_job.calculation_type import CalculationType
+
+DEFAULT_REPORT_ID = "the-report-id"
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/settlement_report_job/test_settlement_report_args.py:28]: Must it be a UUID?

File: source/databricks/calculation_engine/package/settlement_report_job/logger.py
Comment (JesperGomes) [reply source/databricks/calculation_engine/package/settlement_report_job/logger.py:1]: I have added it as AC to this story:
https://app.zenhub.com/workspaces/taskforce-settlement-reports-66d955bb0ac43f0036e87e5c/issues/gh/energinet-datahub/green-energy-hub/633

File: source/databricks/calculation_engine/package/settlement_report_job/settlement_report.py (hunk 1 above)
Comment (JesperGomes) [reply source/databricks/calculation_engine/package/settlement_report_job/settlement_report.py:45]: done

File: source/databricks/calculation_engine/tests/settlement_report_job/conftest.py (hunk 2 above)
Comment (JesperGomes) [reply source/databricks/calculation_engine/tests/settlement_report_job/conftest.py:28]: oops

File: source/databricks/calculation_engine/package/settlement_report_job/args_helper.py
Comment (JesperGomes) [reply source/databricks/calculation_engine/package/settlement_report_job/args_helper.py:1]: Maybe - I would prefer to wait a bit and evaluate it again later

**Review by BjarkeMeier — APPROVED** [review]

**Review by kerbou — APPROVED** [review]

**Pull request merged**